.github/workflows/open-pr.yml

```diff
@@ -0,0 +1,40 @@
+name: Lint and Type Check - Open PR
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - main
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - uses: pnpm/action-setup@v4
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "lts/*"
+          cache: "pnpm"
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+      - name: Check Linting
+        run: pnpm lint-ci
+  type-check:
+    needs: lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - uses: pnpm/action-setup@v4
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "lts/*"
+          cache: "pnpm"
+      - name: Install dependencies
```

Comment:
Why do we need to install all dependencies again?

Reply:
`actions/setup-node` have built in support for `actions/cache`, when we install dependencies in the first job i.e `lint`, it will cache the dependencies for the subsequent jobs, in the `type-check` it will fetch dependencies from cache instead of downloading them, or it will download if there is a cache miss.